### DISCUSSION TITLE
In-app update

### DISCUSSION THREAD

**Discussion by BeanXOfficial** [f1multiviewer/issue-tracker#39] [Feature requests] [answered]
I'm pretty sure this has been thought about already, but just putting it out there in case it hasn't. 
It would be great to have a window pop-up inside the app if there is a new update, simply click "update" and things take care of itself. 
I think this might be very useful as I can imagine many potential users get intimidated a little bit by the current way of "installing" (especially on windows).

**Comment by f1multiviewer** [comment]
Unfortunately, auto-updates require code signing, which would require me to give up my anonymity, which I'm not willing to do (for now). For Windows, I do hope to have a normal installer soon, so it should be slightly less 'daunting' to install :)

**Reply by FelixSFD** [reply]
would macOS accept self-signed certificates, if we'd explicitly installed that certificate in our keystores? Like Safari would accept HTTPS-certs from your local CA, if you installed the CA-certificate.

**Reply by malob** [reply]
Just wanted to leave a note to say, that the app not being signed on macOS means that it cannot be added to Homebrew: https://github.com/Homebrew/homebrew-cask/pull/132560

**Reply by f1multiviewer** [reply]
Thanks for the heads-up @malob! I'm still working on this, one of the MultiViewer users reached out to possibly let me use their signing certificate (or at least, add me to their organisation). A small note: the number before the file name (the `asset_id`) is actually important, the file name part is basically ignored, you can name it anything. If you don't change the `asset_id` it'll download the same file.

**Reply by IvoElferink** [reply]
Unsigned it's not possible to have it added to the official homebrew-cask repository, but you can create your own homebrew tap and casks to provide unsigned applications as well. I've created my own personal cask for multiviewer-for-f1-beta: https://github.com/IvoElferink/homebrew-ivel/blob/main/Casks/multiviewer-for-f1-beta.rb 

This doesn't solve the signing/security part in macOS, but makes updating with brew much easier. If wanted you can copy/fork/other it and provide it directly from the source :)

**Reply by f1multiviewer** [reply]
Next release will likely have code-signed/notarized builds, so this should be possible then. I'm also planning to add automatic updates as well.

**Reply by f1multiviewer** [reply]
> Unsigned it's not possible to have it added to the official homebrew-cask repository, but you can create your own homebrew tap and casks to provide unsigned applications as well. I've created my own personal cask for multiviewer-for-f1-beta: https://github.com/IvoElferink/homebrew-ivel/blob/main/Casks/multiviewer-for-f1-beta.rb
> 
> This doesn't solve the signing/security part in macOS, but makes updating with brew much easier. If wanted you can copy/fork/other it and provide it directly from the source :)

https://github.com/IvoElferink/homebrew-ivel/blob/main/Casks/multiviewer-for-f1-beta.rb#L5 this line won't work. The number in the path is the file ID, which is vital for selecting the newest version. Just changing the file name doesn't do anything, unfortunately. I'll see if, with a new update server, I can get a simpler URL to get the latest release.

**Reply by IvoElferink** [reply]
Nice finding of the fileid in the used URL, I've updated it. It's normal to update the cask file with the version and sha at line 2 and 3 with new releases, the fileid is now part of the version number.

**Reply by malob** [reply]
FYI, MultiViewer for F1 is now properly a Cask in Hombrew: https://github.com/Homebrew/homebrew-cask/pull/142311

**Reply by IvoElferink** [reply]
@malob Thank you, that's great! I'll stop maintaining my personal tap/cask and will use that one from now on, and if needed/possible I'll be committing updates on it as well :)

**Reply by malob** [reply]
@IvoElferink, great, thanks!

**Comment by double-u-a** [comment]
The Electron docs mention some simple update servers that can be run privately without needing to sign anything https://www.electronjs.org/docs/latest/tutorial/updates/#using-other-update-services

Hopefully that means it can be done cross platform

**Reply by f1multiviewer** [reply]
FYI: https://www.electronjs.org/docs/latest/api/auto-updater#macos

> Note: Your application must be signed for automatic updates on macOS. This is a requirement of Squirrel.Mac.

**Comment by JJWatMyself** [comment]
I think this is now in place but as stated, the manual process is still required.

![image](https://user-images.githubusercontent.com/22590445/178125241-a50d56d9-845b-4a5a-a969-b5170642ce9a.png)

**Comment by LapsTimeOFF** [comment]
#f1MVxed now ! (in beta)

**Comment by f1multiviewer** [comment, marked as answer]
Hah you beat me to it @LapsTimeOFF, but yes, there's a beta release that has this. I'm still making a few changes to ease the upgrade path for everyone, but this should be fixed with the next (stable) release!

**Reply by LapsTimeOFF** [reply]
Hehe

**Comment by Roosta078** [comment]
Linux updates are still not supported.  A low time investment improvement would be to create a permalink to the latest release.  That would enable users to easily write a script to pull the latest version.
